### PR TITLE
[devtools-blog] Introducing Memory Inspector

### PR DESCRIPTION
**Pull request by chybie**

### CODE REVIEW

**Comment by chybie** [comment]
@ktran PTAL

**Comment by ktran** [comment]
LGTM, thanks a lot Jecelyn!

**Review by mathiasbynens — APPROVED** [review]

File: site/en/blog/memory-inspector/index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+{% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/yvA6iafQHoD6WazTmodw.png", alt="Components diagram", width="624", height="394" %}
+
+The `LinearMemoryInspector` adds itself as a listener on the `LinearMemoryNavigator`. The `addressChanged` function is to be triggered on an `address-changed` event. As soon as the user is now editing the address input, this sends out the aforementioned event, such that the `addressChanged` function is called. This function now saves the address internally, and updates its subcomponents using the a `data(address, ..)` setter. The subcomponents save the address internally and re-render their views, showing the content at that particular address.
+
+
+### Design Goal: Making Performance and Memory Consumption Independent of the Buffer Size
```
Comment (mathiasbynens) [inline site/en/blog/memory-inspector/index.md:63]: Sentence case for headers, for consistency with other headers:

```suggestion
### Design goal: making performance and memory consumption independent of the buffer size
```

File: site/en/blog/memory-inspector/index.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+With this information we can ensure that we can still render the same view as before (the content around the `address`), without actually having all the data in place. So what to do if a different address is requested, which falls into a different range? In that case, the `LinearMemoryInspector` triggers a `RequestMemoryEvent`, which updates the current range that is kept; an example is shown below:
+
+ {% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/dn2lqBJdF84zIv45KsFL.png", alt="Event trigger flow diagram", width="624", height="180" %}
+
+In this example, the user navigates the memory page (the Memory Inspector is using paging for showing chunks of data), which triggers a `PageNavigationEvent`, which itself triggers a `RequestMemoryEvent`.
+That event kicks off fetching the new range, which is then propagated to the `LinearMemoryInspector` component through setting the data. As a result, we show the newly fetched data.
+
+
+## Oh, and did you know? You can even inspect memory in Wasm and C/C++ code
```
Comment (mathiasbynens) [inline site/en/blog/memory-inspector/index.md:83]:
```suggestion
## Oh, and did you know? You can even inspect memory in Wasm and C/C++ code

```

File: site/en/blog/memory-inspector/index.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+<!-- {% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/GcHy7qArrsUhF8UaJVIT.png", alt="Inspect memory in Wasm", width="800", height="441" %} -->
+{% Glitch {
+  id: 'memory-inspector',
+  path: 'demo-cpp.html',
+  height: 700
+} %}
+
+
+## Conclusion
```
Comment (mathiasbynens) [inline site/en/blog/memory-inspector/index.md:94]:
```suggestion
## Conclusion

```

File: site/en/blog/memory-inspector/index.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+With Chrome 91 we are introducing the Memory Inspector, a tool to inspect array buffers. You might have previously seen memory inspection tools to view binary data, which show the binary content in a grid along with their addresses, and which offer different ways of interpreting the underlying values. This is what the Memory Inspector is bringing to you. With the Memory Inspector you can now view the content, navigate it, and select types to be used to interpret the values at hand. It shows the ASCII values directly next to the bytes, and it allows the user to select different endianness. See the Memory Inspector in action below:
+
+{% Video src="video/dPDCek3EhZgLQPGtEG3y0fTn4v82/gwh42P11Rat7evNHqzfg.mp4", muted="true", controls="true" %}
+
+Want to try it out? To learn how to open the Memory Inspector and view your array buffer (or TypedArray, DataView, or Wasm Memory) and more information on how to use it, head over to [our documentation on the Memory Inspector](/docs/devtools/memory-inspector/). Give it a try on [these toy examples](http://memory-inspector.glitch.me/) (for JS, Wasm, and C++).
+
+
+## Designing the Memory Inspector
```
Comment (mathiasbynens) [inline site/en/blog/memory-inspector/index.md:41]:
```suggestion
## Designing the Memory Inspector

```

**Review by bmeurer — APPROVED** [review]

**Pull request merged**